src/Container/AbstractEventStoreFactory.php

```diff
@@ -94,15 +81,15 @@ public function __invoke(ContainerInterface $container): EventStore
         if (isset($config['connection_service'])) {
             $connection = $container->get($config['connection_service']);
         } else {
-            $separator = $this->driverSchemeSeparators[$config['connection_options']['driver']];
-            $dsn = $this->driverSchemeAliases[$config['connection_options']['driver']] . ':';
-            $dsn .= 'host=' . $config['connection_options']['host'] . $separator;
-            $dsn .= 'port=' . $config['connection_options']['port'] . $separator;
-            $dsn .= 'dbname=' . $config['connection_options']['dbname'] . $separator;
-            $dsn = rtrim($dsn);
-            $user = $config['connection_options']['user'];
-            $password = $config['connection_options']['password'];
-            $connection = new PDO($dsn, $user, $password);
+            $connection = new PDO(
+                $this->buildConnectionDsn($config['connection_options']),
+                $config['connection_options']['user'],
+                $config['connection_options']['password']
+            );
+
+            if (isset($config['connection_options']['charset'])) {
+                $connection->query("SET NAMES '".$config['connection_options']['charset']."'");
```

Comment:
The connection option for charset is used in mysql factory. This is a workaround for postgre.